src/api/services/contentlet-service.ts

```diff
@@ -7,17 +7,42 @@ import {ApiRoot} from '../persistence/ApiRoot';
 import {Http} from '@angular/http';
 import {LoginService} from './login-service';
 import {Subject} from 'rxjs/Subject';
+import {DotcmsEventsService} from './dotcms-events-service';
 
 @Injectable()
 export class ContentletService extends CoreWebService {
-
+    private structureTypeView: StructureTypeView[];
     private _structureTypeView$: Subject<StructureTypeView[]> = new Subject<StructureTypeView[]>();
 
-    constructor(apiRoot: ApiRoot, http: Http, loginService: LoginService) {
+    constructor(apiRoot: ApiRoot, http: Http, loginService: LoginService, dotcmsEventsService: DotcmsEventsService) {
         super(apiRoot, http);
 
         loginService.watchUser(this.loadContentTypes.bind(this));
-        loginService.logoutAs().subscribe( () => this.loadContentTypes());
+
+        dotcmsEventsService.subscribeTo('SAVE_BASE_CONTENT_TYPE').pluck('data').subscribe( contentTypeView => {
```

Comment:
This is pretty cool and extensible (liked)
The only thing I can see is that we should introduce a logging library, the console.log should be not showed on production mode, just on developer mode.

Thanks